ArcGISRuntimeSDKQt_CppSamples/Maps/ShowDeviceLocationUsingIndoorPositioning/ShowDeviceLocationUsingIndoorPositioning.cpp

```diff
@@ -0,0 +1,132 @@
+  const QStringList globalIdSortNames = {"DateCreated", "DATE_CREATED"};
+
+  const QStringList layerNames = {"Details", "Units", "Levels"};
+}
+
+ShowDeviceLocationUsingIndoorPositioning::ShowDeviceLocationUsingIndoorPositioning(QObject* parent /* = nullptr */):
+  QObject(parent)
+{
+  Credential* cred = new Credential(username, password, this);
+  Portal* portal = new Portal(portalUrl, cred, this);
+  m_map = new Map(new PortalItem(portal, itemId, this), this);
+}
+
+ShowDeviceLocationUsingIndoorPositioning::~ShowDeviceLocationUsingIndoorPositioning() = default;
+
+void ShowDeviceLocationUsingIndoorPositioning::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<ShowDeviceLocationUsingIndoorPositioning>("Esri.Samples", 1, 0, "ShowDeviceLocationUsingIndoorPositioningSample");
+}
+
+MapQuickView* ShowDeviceLocationUsingIndoorPositioning::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void ShowDeviceLocationUsingIndoorPositioning::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  setupIndoorsLocationDataSource();
+
+  emit mapViewChanged();
+}
+
+// This function uses a helper class `IndoorsLocationDataSourceCreator` to construct the IndoorsLocationDataSource
+void ShowDeviceLocationUsingIndoorPositioning::setupIndoorsLocationDataSource()
+{
+  IndoorsLocationDataSourceCreator* indoorsLocationDataSourceCreator = new IndoorsLocationDataSourceCreator(this);
+
+  connect(indoorsLocationDataSourceCreator, &IndoorsLocationDataSourceCreator::createIndoorsLocationDataSourceCompleted, this, [this](IndoorsLocationDataSource* indoorsLDS)
+  {
+    connect(m_mapView->locationDisplay(), &LocationDisplay::locationChanged, this, &ShowDeviceLocationUsingIndoorPositioning::locationChangedHandler);
+
+    m_mapView->locationDisplay()->setDataSource(indoorsLDS);
+    m_mapView->locationDisplay()->setAutoPanMode(LocationDisplayAutoPanMode::Navigation);
+    m_mapView->locationDisplay()->start();
+  });
+
+  indoorsLocationDataSourceCreator->createIndoorsLocationDataSource(m_map, positioningTableName, pathwaysLayerName);
+}
+
+// Change currently displayed location information and change floor display if necessary
+void ShowDeviceLocationUsingIndoorPositioning::locationChangedHandler(Location loc)
+{
+  if (m_locationProperties["floor"] != m_currentFloor)
+  {
+    m_currentFloor = m_locationProperties["floor"].toInt();
+    changeFloorDisplay();
+  }
+  m_locationProperties = loc.additionalSourceProperties();
+  m_locationProperties.insert("horizontalAccuracy", loc.horizontalAccuracy());
+  qDebug() << m_locationProperties;
+
+  emit locationChanged();
+}
+
+void ShowDeviceLocationUsingIndoorPositioning::changeFloorDisplay()
+{
+  const LayerListModel* layers = m_map->operationalLayers();
+  for (Layer* layer : *layers)
```

Comment:
If we don't instantiate the layer list model variable, the IDE complains that the qlist may detach, so this is a best practice in this regard

Reply:
That IDE warning could be bogus. There is no QList with that syntax.

If it's this warning, it's wrong.
https://github.com/KDE/clazy/blob/master/docs/checks/README-range-loop-detach.md

We frequently use `*(m_map->operationalLayers())` and it should be fine with no detach or performance concerns.

I mention this because I've seen clazy get things wrong in some cases. Not sure if that applies here.